packages/graph/hash_graph/lib/graph/src/shared/identifier/time/timestamp.rs

```diff
@@ -0,0 +1,113 @@
+use core::fmt;
+use std::{any::type_name, error::Error, marker::PhantomData, time::SystemTime};
+
+use chrono::{DateTime, Utc};
+use derivative::Derivative;
+use postgres_types::{private::BytesMut, FromSql, ToSql, Type};
+use serde::{Deserialize, Serialize};
+use utoipa::{openapi, ToSchema};
+
+/// Opaque structure to represent a single point in time.
+///
+/// The type parameter `A` is the time axis to distinguish between different time axis at compile
+/// time.
+// A generic parameter is used here to avoid implementing the same struct multiple times or using
+// macros. It's reused in other time-related structs as well. This implies that trait bounds are
+// not required for trait implementations.
+#[derive(Derivative, Serialize, Deserialize)]
+#[derivative(
+    Copy(bound = ""),
+    Clone(bound = ""),
+    PartialEq(bound = ""),
+    Eq(bound = ""),
+    Hash(bound = ""),
+    PartialOrd(bound = ""),
+    Ord(bound = "")
+)]
+#[serde(transparent, bound = "")]
+pub struct Timestamp<A> {
+    #[serde(skip)]
+    axis: PhantomData<A>,
+    time: DateTime<Utc>,
+}
+
+impl<A> fmt::Debug for Timestamp<A> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.debug_struct("timestamp")
+            .field("axis", &type_name::<A>())
```

Comment:
Does this stop us using `derivative(Debug..`?

Reply:
Yes. I mainly implemented it this way to allow `Debug(bound = "")` on structs, which are using `Timestamp`.